lib/functions/configuration/main-config.sh

```diff
@@ -9,6 +9,24 @@
 
 ### Attention: we can't use any interactive programs, read from stdin, nor use non-coreutils utilities here.
 
+function pre_install_distribution_specific__disable_cnf_apt_hook(){
+	if [ ! "${ARCH}" == amd64 ] &&  [ -f "${SDCARD}"/etc/apt/apt.conf.d/50command-not-found ]; then #disable command-not-found (60% build-time saved under qemu)
```

Comment:
Hmm. This is not true when building arm64 on amd64, which is also slow. Search for "native build" and you'll find the correct comparison (host_arch != target_arch)

Reply:
got the point, you're right - but can't wrap my head around why it's not working: it seems that $host_arch and $target_arch are not initialized when main-config.sh runs - I put a "display_alert" echoing those vars and I get them empty - so comparison always runs false

I need that file to be disable right after 
`[🐳|🌱] Disabling Armbian repo [ arm64-trixie :: skip:no when:image-early ]` to be effective

In line of principle there's also to be considered that command-not-found maps package-files to package-names, and it's completely useless at build-time - while disabling on native build it will shove just 10-17 seconds while on qemu builds will save 10s of minutes, having it unconditional would still provide a faster build

I couldn't find anything for "native build" by grepping lib/ - will keep trying and will get that working: but I admit to be currently slightly lost :)

Reply:
sequencing was the key - I moved function hooks down after `do_main_configuration` and it works - pulling the commit and resolving conversation

Thanks Ricardo for:
- your patience
- your support
- your patience again